[Konsist] Integration & Custom Tests

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
🛠 Need an AI review?  
Add a comment: `@coderabbitai review`
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
-->

---

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

`TODO`

---

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

`TODO`

---

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
Build: Group registered tasks on the root build gradle file

ParaskP7 committed:
Build: Add konsist module

ParaskP7 committed:
Analysis: Add konsist tests

ParaskP7 committed:
CI: Add konsist jon on ci

FYI: This change includes adding this extra script related
'-x libs:konsist-test:testDebugUnitTest' configuration on unit tests so
that 'Konsist' tests are excluded from running as part of all the
other "normal" unit tests.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
Analysis: Convert woopos design system btn/text usage rules to konsist

This commit demonstrates how an existing custom Detekt rule can be
easily converted to Konsist, and, at the same time showcases how
Konsist's declarative nature makes it easier to write and maintain such
checks.

ParaskP7 committed:
Analysis: Convert compose view model guideline to konsist

This commit demonstrates how an existing documented Compose guideline
can be easily converted to Konstist, and, at the same time showcases how
adding a KDoc on top of this custom test/rule make this a living
document.

ParaskP7 commented:
Refer to [pause comment](https://github.com/Automattic/apps-infra-plans/issues/157#issuecomment-2792069226) on [AIP#157](https://github.com/Automattic/apps-infra-plans/issues/157).
